Module4_Labs/Lab3_File_System/Cards/411.md

```diff
@@ -4,6 +4,12 @@
 
 <!--concepts={directedGraphs, introToGraphs, useOfGraphs, Depth First Search (DFS), Stack Manipulation}-->
 
+Let's review `DFS` as a concept before we dive into its application to this specific problem. `DFS` is an algorithm which traverses a graph (directed or undirected) through the use of a data structure called a `stack` (Last-In First-Out). We start with a given start node, mark that it has been visited, and push its unvisited neighbors to the stack. We then pop the top node off of the stack, mark it as visited, and repeat this process until all the nodes have been visited. 
+
```

Comment:
This may not be necessary if the student has access to the DFS concept, but I really like the description you made so I would keep it in there for right now